planning/behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -584,14 +674,15 @@ Pose GoalPlannerModule::calcRefinedGoal(const Pose & goal_pose) const
   return refined_goal_pose;
 }
 
-bool GoalPlannerModule::planFreespacePath()
+bool GoalPlannerModule::planFreespacePath(
+  std::shared_ptr<const PlannerData> planner_data,
+  const std::shared_ptr<OccupancyGridBasedCollisionDetector> occupancy_grid_map)
 {
   GoalCandidates goal_candidates{};
   {
     const std::lock_guard<std::recursive_mutex> lock(mutex_);
-    goal_searcher_->setPlannerData(planner_data_);
     goal_candidates = thread_safe_data_.get_goal_candidates();
-    goal_searcher_->update(goal_candidates);
+    goal_searcher_->update(goal_candidates, occupancy_grid_map, planner_data);
```

Comment:
better to use `goal_searcher = gp_planner_data.goal_searcher;` from `onFreespaceParkingTimer`?

but , `goal_searcher_->update` is const functions, so goal_seacher_ seems also ok.

Reply:
yes, exactly!